libr/anal/fcn.c

```diff
@@ -229,9 +229,14 @@ static int try_walkthrough_jmptbl(RAnal *anal, RAnalFunction *fcn, int depth, ut
 	ut64 offs, sz = anal->bits >> 3;
 	if (!jmptbl) return 0;
 	anal->iob.read_at (anal->iob.io, ptr, jmptbl, MAX_JMPTBL_SIZE);
-	for (offs = 0; offs < MAX_JMPTBL_SIZE; offs += sz) {
-		ut64 jmpptr = 0;
-		r_mem_copyendian ((ut8*)&jmpptr, jmptbl + offs, sz, !anal->big_endian);
+	for (offs = 0; offs + sz - 1 < MAX_JMPTBL_SIZE; offs += sz) {
```

Comment:
Grmpf :[[

> On 04 May 2016, at 09:59, Damien Zammit <notifications@github.com> wrote:
> 
> In libr/anal/fcn.c:
> 
> > @@ -229,9 +229,14 @@ static int try_walkthrough_jmptbl(RAnal *anal, RAnalFunction *fcn, int depth, ut
> >  	ut64 offs, sz = anal->bits >> 3;
> >  	if (!jmptbl) return 0;
> >  	anal->iob.read_at (anal->iob.io, ptr, jmptbl, MAX_JMPTBL_SIZE);
> > -	for (offs = 0; offs < MAX_JMPTBL_SIZE; offs += sz) {
> > -		ut64 jmpptr = 0;
> > -		r_mem_copyendian ((ut8*)&jmpptr, jmptbl + offs, sz, !anal->big_endian);
> > +	for (offs = 0; offs + sz - 1 < MAX_JMPTBL_SIZE; offs += sz) {
> I tried reading in opposite endian and it made no difference to the mips branch detection test.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
>